### PR TITLE
[webview_flutter] Support new onUrlChanged event on iOS.

### PR DESCRIPTION
**Pull request by BeMacized**
This PR adds iOS support for the new onUrlChanged event added to the platform interface in #4509.

As this PR contains and depends on the changes made to the platform interface in #4509, this PR will remain in draft state until that PR has been successfully merged and the changes have been published.

Relevant issue: 
- flutter/flutter#27729

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FLTWKNavigationDelegate.m [hunk 1]
```diff
@@ -16,6 +16,19 @@ - (instancetype)initWithChannel:(FlutterMethodChannel *)channel {
...
 
+- (void)observeValueForKeyPath:(NSString *)keyPath
+                      ofObject:(id)object
+                        change:(NSDictionary<NSKeyValueChangeKey, id> *)change
+                       context:(void *)context {
+  if (![keyPath isEqualToString:@"URL"]) {
+    return;
+  }
+  NSURL *newURL = [change objectForKey:NSKeyValueChangeNewKey];
+  if (newURL) {
```
Comment (mvanbeusekom) [inline packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FLTWKNavigationDelegate.m:27]: There should be a unit test verifying that the method channel is not invoked when the `newUrl` parameter is `nil`.

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]

File: packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FLTWKNavigationDelegate.m (hunk 1 above)
Comment (BeMacized) [reply packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FLTWKNavigationDelegate.m]: Added

**Comment by Hixie** [comment]
@BeMacized are you still interested in working on this PR? Is there anything you need from anyone?

**Comment by BeMacized** [comment]
@Hixie Same as in the other PR: We stopped working on this feature due to there being no good way to implement it in the current structure without a breaking change. The changes to the platform interface these PRs depend on that were merged have been reverted. Once [#94051](https://github.com/flutter/flutter/issues/94051) has been resolved I'll be happy to pick it back up again.

**Comment by Hixie** [comment]
@BeMacized How about this one?

**Pull request closed without merging**